### DISCUSSION TITLE
weblate_default_password

### DISCUSSION THREAD

**Discussion by rbos** [WeblateOrg/weblate#7971] [General]
If I remove WEBLATE_DEFAULT_PASSWORD from docker.override.yml in the Weblate configuration, will it randomize the admin password every time it starts up Docker containers?

The documentation says "If WEBLATE_ADMIN_PASSWORD is not set, the admin user is created with a random password shown on first startup."

What constitutes first startup? Does a new version count as a "first startup"? Docker recreates containers a lot, so I'm unclear.

My goal is to have a persistent admin password without coding it in a text file.

**Comment by nijel** [comment]
I guess you are talking about [WEBLATE_ADMIN_PASSWORD](https://docs.weblate.org/en/latest/admin/install/docker.html#envvar-WEBLATE_ADMIN_PASSWORD).

If the admin user does not exist, the user is created with a random password. If it exists, it is not generated again.

**Reply by nijel** [reply]
Pull requests to clarify the documentation are welcome ;-).
